packages/E2ETest/windows/TreeDumpLibrary/TreeDumpControlViewManager.cs

```diff
@@ -0,0 +1,228 @@
+            if (m_uiaID != null)
+            {
+                var matchingNode = FindChildWithMatchingUIAID(current);
+                if (matchingNode != null)
+                {
+                    dumpRoot = matchingNode;
+                }
+            }
+
+            String dumpText = VisualTreeDumper.DumpTree(dumpRoot, m_textBlock /* exclude */);
+            if (dumpText != m_dumpExpectedText)
+            {
+                await MatchDump(dumpText);
+            }
+        }
+
+        private async Task MatchDump(string dumpText)
+        {
+            if (m_dumpExpectedText == null)
+            {
+                try
+                {
+                    var file = await Windows.ApplicationModel.Package.Current.InstalledLocation.GetFileAsync(@"Assets\TreeDump\" + m_dumpID + ".txt");
+                    m_dumpExpectedText = await Windows.Storage.FileIO.ReadTextAsync(file);
+                }
+                catch (IOException)
+                {
+                    UpdateResult(false /*matchDump*/, "Tree dump master file not found in testapp package!");
+                }
+            }
+
+            if (m_dumpExpectedText != dumpText)
+            {
+                StorageFolder storageFolder = ApplicationData.Current.LocalFolder;
+                string fileName = "TreeDump\\" + m_dumpID + ".out";
+                try
+                {
+                    StorageFile outFile = await storageFolder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
+                    await Windows.Storage.FileIO.WriteTextAsync(outFile, dumpText);
+                    UpdateResult(false /*matchDump*/, "Tree dump file does not match master! See output at " + outFile.Path);
+                }
+                catch (IOException)
+                {
+                    UpdateResult(false /*matchDump*/, "Can't write dump output file:" + fileName);
+                }
+            }
+            else
+            {
+                UpdateResult(true /*matchDump*/, "");
+            }
+        }
+
+        private async void UpdateResult(bool matchDump, string helpText)
+        {
+            if (matchDump)
+            {
+                UpdateTextBlockText("TreeDump:Passed");
+            }
+            else
+            {
+                UpdateTextBlockText("TreeDump:Failed, click to see more!");
+                m_errString += "\r\n" + helpText;
+
+                StorageFolder storageFolder = ApplicationData.Current.LocalFolder;
+                string fileNameError = "TreeDump\\" + m_dumpID + ".err";
+                try
+                {
+                    StorageFile errFile = await storageFolder.CreateFileAsync(fileNameError, CreationCollisionOption.ReplaceExisting);
+                    await Windows.Storage.FileIO.WriteTextAsync(errFile, m_errString);
+                }
+                catch (IOException)
+                {
```

Comment:
But we are inside writing out the error text file, if that fails (i.e. file is opened in Notepad) I don't know where else to log...:)

---
In reply to: [357860121](https://github.com/microsoft/react-native-windows/pull/3754#discussion_r357860121) [](ancestors = 357860121)

Reply:
Looks like I should use GenerateUniqueName collision option to avoid issues when file is currently opened already.

---
In reply to: [357875080](https://github.com/microsoft/react-native-windows/pull/3754#discussion_r357875080) [](ancestors = 357875080,357860121)